git-webui/src/share/git-webui/webui/js/git-webui.js

```diff
@@ -1895,8 +1895,14 @@ webui.DiffView = function(sideBySide, hunkSelectionAllowed, parent, stashedCommi
             return;
         }
         var stashIndex = parseInt($(".log-entry.active .stash-list-index").text());
-        webui.git("stash apply stash@{"+stashIndex+"}", function(output){
-            webui.showSuccess(output);
+        $.post("git", {command: "stash apply stash@{"+stashIndex+"}"}, function(output) {
+            if (output.substring(0,5) == "ERROR") {
```

Comment:
Would rather keep cleaner API for communicating error status (using HTTP error codes / returnCode w/ standardized handling of an error response)